File: Data Structures & Algorithms/buy-and-sell-crypto/submission-4.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        maxP = 0
        minP = prices[0]

        min = 10000
        max = -1

        for i in range(len(prices)):
            if prices[i] < min:
                min = prices[i]
                max = prices[i]

            if prices[i] > max:
                max = prices[i]

        
        return max - min

File: Data Structures & Algorithms/max-water-container/submission-4.cpp
class Solution {
public:
    int maxArea(vector<int>& heights) {
        int l = 0;
        int r = heights.size() - 1;
        int max_area = min(heights[l], heights[r]) * (r - l);

        while (l < r) {
            if (heights[l] < heights[r]) {
                l += 1;
            } else {
                r -= 1;
            }

            int val = min(heights[l], heights[r]) * (r - l);
            max_area = max(val, max_area);
        }

        return max_area;
    }
};